kernel/src/common/deferred_call_mux.rs

```diff
@@ -0,0 +1,207 @@
+//! Hardware-independent kernel interface for deferred calls
+//!
+//! This allows any struct in the kernel which implements
+//! [DeferredCallMuxClient](crate::common::deferred_call_mux::DeferredCallMuxClient)
+//! to set and receive deferred calls.
+//!
+//! These are scheduled "interrupts" in the chip scheduler.
+//! They are especially important if some hardware doesn't
+//! support real interrupts where they are needed, or for
+//! implementing software devices that are supposed to work
+//! like hardware devices.
+//!
+//! Usage
+//! -----
+//!
+//! The `deferred_call_mux_clients` array size determines how many
+//! [DeferredCallHandle](crate::common::deferred_call_mux::DeferredCallHandle)s
+//! may be registered with the Mux.
+//! When no more slots are available,
+//! `deferred_call_mux.register(some_client)` will return `None`.
+//!
+//! ```
+//! # use core::cell::Cell;
+//! # use kernel::common::cells::OptionalCell;
+//! # use kernel::static_init;
+//! use kernel::common::deferred_call_mux::{
+//!     DeferredCallMux,
+//!     DeferredCallMuxClient,
+//!     set_global_mux,
+//! };
+//!
+//! let deferred_call_mux_clients = unsafe { static_init!(
+//!     [(Cell<bool>, OptionalCell<&'static DeferredCallMuxClient>); 1],
+//!     [(Cell::new(false), OptionalCell::empty())]
+//! ) };
+//! let deferred_call_mux = unsafe { static_init!(
+//!     DeferredCallMux,
+//!     DeferredCallMux::new(deferred_call_mux_clients)
+//! ) };
+//! assert!(unsafe { set_global_mux(deferred_call_mux) }, true);
+//!
+//! # struct SomeCapsule;
+//! # impl SomeCapsule {
+//! #     pub fn new(_mux: &'static DeferredCallMux) -> Self { SomeCapsule }
+//! #     pub fn set_deferred_call_handle(
+//! #         &self,
+//! #         _handle: kernel::common::deferred_call_mux::DeferredCallHandle,
+//! #     ) { }
+//! # }
+//! # impl DeferredCallMuxClient for SomeCapsule {
+//! #     fn call(
+//! #         &self,
+//! #         _handle: kernel::common::deferred_call_mux::DeferredCallHandle,
+//! #     ) { }
+//! # }
+//! #
+//! // Here you can register custom capsules, etc.
+//! // This could look like:
+//! let some_capsule = unsafe { static_init!(
+//!     SomeCapsule,
+//!     SomeCapsule::new(deferred_call_mux)
+//! ) };
+//! some_capsule.set_deferred_call_handle(
+//!     deferred_call_mux.register(some_capsule).expect("no deferred call slot available")
+//! );
+//! ```
+
+use crate::common::cells::OptionalCell;
+use crate::common::deferred_call::DeferredCall;
+use core::cell::Cell;
+
+pub const DEFERRED_CALL_MUX_TASK: usize = 31; // highest bit in usize reserved for deferred call mux
```

Comment:
make this comment `rust doc` instead?

Reply:
I've made it into a rustdoc comment. In addition to that, the `DEFERRED_CALL_MUX_TASK` is now always set to the highest bit in a usize, regardless of a 32- or 64-bit architecture.